File: Data Structures & Algorithms/climbing-stairs/submission-0.cpp
class Solution {
public:
    unordered_map<int, int> tab;
    int climbStairs(int n) {
        tab[1] = 1;
        tab[2] = 2;
        for(int i = 3; i <= n; i++)
        {
            tab[i] = tab[i-1] + tab[i-2];
        }
        return tab[n];
    }
};

File: Data Structures & Algorithms/palindromic-substrings/submission-0.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        n = len(s)
        tab = [[False] * n for x in range(n)]
        count = 0

        for i in range(n - 1, -1, -1):
            for j in range(i, n):
                if s[i] == s[j] and (j - i <= 2 or tab[i+1][j-1]):
                    tab[i][j] = True
                    count += 1
        
        return count